test/unit/math/test_ad.hpp

```diff
@@ -1334,6 +1393,27 @@ void expect_ad_vectorized_binary(const ad_tolerances& tols, const F& f,
   expect_ad_vectorized_binary_impl(tols, f, to_std_vector(x), to_std_vector(y));
 }
 
+/**
+ * Test that the specified vectorized polymorphic binary function
+ * produces autodiff results consistent with values determined by
+ * double and integer inputs and 1st-, 2nd-, and 3rd-order derivatives
+ * consistent with finite differences of double inputs.
+ *
+ * @tparam F type of polymorphic, vectorized functor to test
+ * @tparam T1 type of first argument
+ * @tparam T1 type of second argument
+ * @param tols tolerances for test
+ * @param f functor to test
+ * @param x value to test
+ * @param y value to test
+ */
+template <typename F, typename T1, typename T2,
+          require_any_std_vector_t<T1, T2>* = nullptr>
```

Comment:
Do any of these things need to be std::vectors?

The two things exposed here are:
```
template <typename F, typename T1, typename T2,
          require_all_eigen_col_vector_t<T1, T2>* = nullptr>
void expect_ad_vectorized_binary(const ad_tolerances& tols, const F& f,
                                 const T1& x, const T2& y)
```

```
template <typename F, typename T1, typename T2,
          require_any_std_vector_t<T1, T2>* = nullptr>
void expect_ad_vectorized_binary(const ad_tolerances& tols, const F& f,
                                 const T1& x, const T2& y)
```

Should two row_vectors work?

Edit: I guess this goes to the other question. Do we intend for this interface to work with `row_vectors` at the Stan level? Not saying it has too -- just trying to understand the extent of the vectorization here.

Reply:
> Should two row_vectors work?


Yep, my original goal was to have the same testing that was used in the ```prim``` vectorisation tests, which test:
```
f(vector, vector)
f(row_vector, row_vector)
f(matrix, matrix)
f(array, array)    // Eigen::Array
f(expr, expr)
f(std::vector, std::vector)
```

Then within each of those combinations, it tests having nested arrays of the inputs, and reals with the inputs. But with the ```expect_ad``` framework, the time for testing got pretty long so I settled for testing:
```
f(vector, vector)
f(std::vector, std::vector)
```

And then combinations of nested arrays and reals with those inputs.

> Should two row_vectors work?

Yep, works like a charm